### ISSUE TITLE
Error when destroying AWS lambda function resources

### ISSUE THREAD

**Issue by jcollado** [hashicorp/waypoint#1641]
**Describe the bug**
Error on `waypoint destroy` when following the AWS lambda deployment tutorial

**Steps to Reproduce**
Follow the tutorial here:
https://learn.hashicorp.com/tutorials/waypoint/aws-lambda

When running `waypoint destroy`, the error is as follows:
```
$ waypoint destroy -auto-approve

» Destroying releases for application 'learn-waypoint-lambda'...

» Destroying deployments for application 'learn-waypoint-lambda'...
! Error destroying: rpc error: code = Unknown desc = ResourceInUse: Target group
'arn:aws:elasticloadbalancing:us-west-2:188077105376:targetgroup/learn-waypoint-lambda-01F7X1X4MA/1bab643a591c9fe5' is currently in use by a listener or a rule
        status code: 400, request id: 835cbdef-d3d6-4fc7-a301-62edd6ed8002
```

On a second execution of the same command, it seems to work fine:
```
$ waypoint destroy -auto-approve

» Destroying shared deploy resources for application 'learn-waypoint-lambda'...
✓ Connecting to AWS
✓ Deleting Lambda function
✓ IAM role deleted
Destroy successful!
```

However, looking at the AWS console, the following resources still exists:
- Application Load Balancer
- Target Group
- Elastic Container Repository

**Expected behavior**
Resources created using `waypoint up` are destroyed.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.4
* Waypoint Server Platform and Version: `docker` 20.10.2 (both client and server)
* Waypoint Plugin: `aws-alb`

**Additional context**
Nothing to add.

**Comment by krantzinator** [comment]
Thanks for opening an issue! We'll be working on this for an upcoming release.

**Comment by evanphx** [comment]
This will be fixed when #1611 is merged.

**Comment by catsby** [comment]
Hello there! https://github.com/hashicorp/waypoint/pull/1648 should resolve this issue, and will go out in the next release. I'm going to close this for now, thanks!
